packages/application-components/src/components/dialogs/internals/dialog-test-utils.tsx

```diff
@@ -0,0 +1,115 @@
+import { ComponentType, Dispatch, ReactNode, useState } from 'react';
+import SecondaryButton from '@commercetools-uikit/secondary-button';
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import { screen, renderComponent, fireEvent } from '../../../test-utils';
+import { TConfirmationDialogProps } from '../confirmation-dialog';
+import { TFormDialogProps } from '../form-dialog';
+import { TInfoDialogProps } from '../info-dialog';
+
+type DialogControllerProps = {
+  children: (renderProps: {
+    isOpen: boolean;
+    setIsOpen: Dispatch<boolean>;
+  }) => JSX.Element;
+};
+const DialogController = (props: DialogControllerProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <div>
+      <SecondaryButton
+        label="Open Info Dialog"
+        onClick={() => setIsOpen(true)}
+      />
+      {props.children({ isOpen, setIsOpen })}
+    </div>
+  );
+};
+DialogController.displayName = 'DialogController';
+
+type TDialogCustomTitleProps = {
+  title: string;
+};
+export const DialogCustomTitle = (props: TDialogCustomTitleProps) => {
+  return (
+    <Spacings.Inset scale="xl">
+      <Text.Headline as="h3" title={props.title} truncate>
+        {props.title}
+      </Text.Headline>
+      <SecondaryButton label="Click me" tone="info" />
+    </Spacings.Inset>
+  );
+};
+
+export type TCreateDialogValidatorParams = {
+  component: ComponentType<
+    TConfirmationDialogProps | TFormDialogProps | TInfoDialogProps
+  >;
```

Comment:
Lets's keep it simple

```suggestion
  component: ElementType;
```

Reply:
Thanks a lot for the suggestion 🙇 

Updated here: c31bc90c72b996181b5261a2fd4a9aee3d860509